src/App.tsx

```diff
@@ -1,61 +1,132 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './components/types/Todo';
 
 export const App = () => {
+  const newTodos: Todo[] = todosFromServer.map(todo => ({
+    ...todo,
+    user:
+      usersFromServer.find(
+        userFromServer => todo.userId === userFromServer.id,
+      ) || null,
+  }));
+  const [todos, setTodos] = useState(newTodos);
+  const [title, setTitle] = useState('');
+  const [titleTouched, setTitleTouched] = useState(true);
+  const [titleError, setTitleError] = useState(false);
+  const [user, setUser] = useState(0);
+  const [userTouched, setUserTouched] = useState(true);
+  const [userError, setUserError] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (title.length <= 0 && user === 0) {
+      setTitleError(true);
+      setTitleTouched(false);
+      setUserError(true);
+      setUserTouched(false);
+
+      return;
+    }
+
+    if (title.length <= 0) {
+      setTitleError(true);
+      setTitleTouched(false);
+
+      return;
+    }
+
+    if (user === 0) {
+      setUserError(true);
+      setUserTouched(false);
+
+      return;
+    }
+
+    const todoId: number =
+      todosFromServer.toSorted((a, b) => b.id - a.id)[0].id + 1;
```

Comment:
The method `toSorted` is not a valid JavaScript method. You should use `sort` instead to sort the `todosFromServer` array. For example: `todosFromServer.sort((a, b) => b.id - a.id)[0].id + 1`.